barretenberg/cpp/pil/vm2/poseidon2_hash.pil

```diff
@@ -4,86 +4,91 @@ include "poseidon2_perm.pil";
 // It is **mostly** well-constrained
 namespace poseidon2_hash;
 
+    pol commit sel;
+    sel * (1 - sel) = 0;
+
     #[skippable_if]
     sel = 0;
 
+    // === Input Values ===
+    // Unpadded Length of the inputs to be hashed
+    pol commit input_len;
+    // Compute the IV using the unpadded length. Only used at the start of a new poseidon2 hash
+    pol TWOPOW64 = 18446744073709551616;
+    pol IV = TWOPOW64 * input_len;
+
     // These are the inputs to be hashed this round, we hash chunks of 3 field elements
     pol commit input_0;
     pol commit input_1;
     pol commit input_2;
 
-    // Output of the hash it is matched with the result of the last permutation round
+    // Output of the hash, not permutation round, it is matched with the result of the last permutation round
     pol commit output;
+    // We enforce that the output has to be kept the same per row until we hit the "end" of the computation
+    sel * (1 - LATCH_CONDITION) * (output' - output) = 0;
```

Comment:
Could you add a comment on when do we expect the latch to be on? (e.g., last row of a poseidon or one before last?) and/or move the latch definition (and dependencies) before it's first use.